variables.tf

```diff
@@ -117,6 +135,7 @@ variable "lifecycle_rules" {
     })
     condition = map(string)
   }))
-  default     = []
   description = "List of lifecycle rules to configure. Format is the same as described in provider documentation https://www.terraform.io/docs/providers/google/r/storage_bucket.html#lifecycle_rule except condition.matches_storage_class should be a comma delimited string."
+  type        = list(string)
```

Comment:
There is already a type constraint defined above actually.

Reply:
whoops! let me take care of that.